### PR TITLE
[feat] adding linodeFirewall to clusters

### PR DESCRIPTION
**Pull request by prajwalvathreya**
**What this PR does / why we need it**: This PR updates the templates in-order to add a firewall by default to clusters of different flavors.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/infra/linodeFirewall.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: LinodeFirewall
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
+  name: ${CLUSTER_NAME}
+spec:
+  enabled: ${LINODE_FIREWALL_ENABLED:=false}
```
Comment (rahulait) [inline templates/infra/linodeFirewall.yaml:9]:
```suggestion
  enabled: ${LINODE_FIREWALL_ENABLED:=false}
  inboundPolicy: DROP
```

Comment (prajwalvathreya) [reply templates/infra/linodeFirewall.yaml:9]: I've added this to the updated template

File: templates/infra/linodeMachineTemplate.yaml
Comment (amold1) [inline templates/infra/linodeMachineTemplate.yaml:1]: Please reset the comment lines to avoid unnecessary formatting changes

File: templates/infra/linodeFirewall.yaml
Comment (amold1) [inline templates/infra/linodeFirewall.yaml:1]: Please add newline at EOF

File: templates/infra/kustomization.yaml
Comment (amold1) [inline templates/infra/kustomization.yaml:1]: Re-add the newline at EOF

**Review by amold1 — CHANGES_REQUESTED** [review]

**Review by amold1 — DISMISSED** [review]
LGTM

**Review by amold1 — APPROVED** [review]
